frontend/src/console/actions.ts

```diff
@@ -70,3 +111,57 @@ export const listUserInstances = (): OcimThunkAction<void> => async dispatch =>
       });
     });
 };
+
+export const updateFieldValue = (
+  instanceId: number,
+  fieldName: string,
+  value: string
+): OcimThunkAction<void> => async dispatch => {
+  dispatch({
+    type: Types.UPDATE_INSTANCE_INFO,
+    fieldName
+  });
+
+  setTimeout(function() {
+    dispatch({
+      type: Types.UPDATE_INSTANCE_INFO_SUCCESS,
+      data: {
+        [fieldName]: value
```

Comment:
This action will get dispatched after some delay (while waiting for the API response), and the `activeInstance` may have changed in the meantime. So you should check and make sure that `getState().activeInstance.id === instanceId` before you dispatch `UPDATE_INSTANCE_INFO_SUCCESS`.

Otherwise if the user submit an update then changes the active instance, then the API response comes in, the user will see the new (title or whatever) on the wrong instance in the UI.

Reply:
Thanks for catching that! There's probably a few places that'll need something like this before dispatching the signal.